### PR TITLE
chore(skills): convert review-edge-case-hunter.xml to native skill

### PR DESCRIPTION
**Pull request by alexeyv**
## Summary

- Replace single-file XML task `review-edge-case-hunter.xml` with native skill directory (`SKILL.md` + `workflow.md` + `bmad-skill-manifest.yaml`), following the pattern from `bmad-review-adversarial-general`
- Update all 5 reference locations: central manifest, module-help.csv, step-04-review.md, and bmad-os-review-pr instructions
- Net zero line change (76 added, 76 removed) — behavior-preserving conversion

## Test plan

- [ ] Verify `review-edge-case-hunter.xml` no longer exists in repo
- [ ] Verify new skill directory has exactly 3 files (SKILL.md, workflow.md, bmad-skill-manifest.yaml)
- [ ] Grep for `review-edge-case-hunter.xml` — zero matches expected
- [ ] Verify workflow.md preserves all 4 JSON output fields and exhaustive enumeration mandate
- [ ] Run installer and confirm skill resolves correctly

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: .claude/skills/bmad-os-review-pr/prompts/instructions.md [hunk 1]
```diff
@@ -130,7 +130,7 @@ Likely tag:
 
 ### 1.2 Run Edge Case Hunter (subagent)
 
-Spawn a subagent that executes the task defined in `_bmad/core/tasks/review-edge-case-hunter.xml`. Pass the full PR diff as the `content` input. Omit `also_consider` unless the user specified extra focus areas.
+Spawn a subagent that invokes the `bmad-review-edge-case-hunter` skill. Pass the full PR diff as the `content` input. Omit `also_consider` unless the user specified extra focus areas.
```
[the remark this answers is not in the document]
Comment (alexeyv) [reply .claude/skills/bmad-os-review-pr/prompts/instructions.md:133]: Fixed in 8be8123f — changed to "The skill returns".

File: src/bmm/workflows/bmad-quick-flow/bmad-quick-dev-new-preview/steps/step-04-review.md [hunk 2]
```diff
@@ -31,7 +30,7 @@ Do NOT `git add` anything — this is read-only inspection.
 **Plan-code-review:** Launch three subagents without conversation context. If no sub-agents are available, generate three review prompt files in `{implementation_artifacts}` — one per reviewer role below — and HALT. Ask the human to run each in a separate session (ideally a different LLM) and paste back the findings.
 
 - **Blind hunter** — receives `{diff_output}` only. No spec, no context docs, no project access. Invoke via the `bmad-review-adversarial-general` skill.
-- **Edge case hunter** — receives `{diff_output}` and read access to the project. Invoke via `{edge_case_hunter_task}`.
+- **Edge case hunter** — receives `{diff_output}` and read access to the project. Invoke via the `bmad-review-edge-case-hunter` skill.
 - **Acceptance auditor** — receives `{diff_output}`, `{spec_file}`, and read access to the project. Must also read the docs listed in `{spec_file}` frontmatter `context`. Checks for violations of acceptance criteria, rules, and principles from the spec and context docs.
```
[the remark this answers is not in the document]
Comment (alexeyv) [reply src/bmm/workflows/bmad-quick-flow/bmad-quick-dev-new-preview/steps/step-04-review.md:34]: Dismissed — the fallback names each skill explicitly so an agent can read the skill workflows. The role descriptions are pointers, not the contract.

File: src/core/tasks/bmad-review-edge-case-hunter/SKILL.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+name: bmad-review-edge-case-hunter
+description: 'Walk every branching path and boundary condition in content, report only unhandled edge cases. Orthogonal to adversarial review - method-driven not attitude-driven.'
+---
```
[the remark this answers is not in the document]
Comment (alexeyv) [reply src/core/tasks/bmad-review-edge-case-hunter/SKILL.md:4]: Rejected — SKILL.md frontmatter is the source of truth for descriptions, not the central manifest. No drift issue.

File: src/core/tasks/bmad-review-edge-case-hunter/workflow.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  "potential_consequence": "what could actually go wrong (max 15 words)"
+}]
+```
+
+No extra text, no explanations, no markdown wrapping. An empty array `[]` is valid when no unhandled paths are found.
+
+
+## HALT CONDITIONS
+
+- If content is empty or cannot be decoded as text, return empty array `[]` and stop
```
[the remark this answers is not in the document]
Comment (alexeyv) [reply src/core/tasks/bmad-review-edge-case-hunter/workflow.md]: Rejected — skill directories are discovered via SKILL.md + bmad-skill-manifest.yaml, not as standalone workflows. collectSkills claims the directory before getWorkflowsFromPath runs.

File: src/core/tasks/bmad-review-edge-case-hunter/workflow.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
+# Edge Case Hunter Review
+
+**Goal:** You are a pure path tracer. Never comment on whether code is good or bad; only list missing handling.
+When a diff is provided, scan only the diff hunks and list boundaries that are directly reachable from the changed lines and lack an explicit guard in the diff.
+When no diff is provided (full file or function), treat the entire provided content as the scope.
+Ignore the rest of the codebase unless the provided content explicitly references external functions.
```
[the remark this answers is not in the document]
Comment (alexeyv) [reply src/core/tasks/bmad-review-edge-case-hunter/workflow.md:6]: Dismissed — guard-in-diff scoping is intentional conservative design. The downstream classify step catches false positives, and most guards fall within diff context window.

[the remark this answers is not in the document]
Comment (alexeyv) [reply src/core/tasks/bmad-review-edge-case-hunter/workflow.md:6]: Dismissed — deleted guards are already covered by "scan diff hunks" and "lack an explicit guard in the diff." Minus lines are part of diff hunks by definition.

File: src/core/tasks/bmad-review-edge-case-hunter/workflow.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+**Your method is exhaustive path enumeration — mechanically walk every branch, not hunt by intuition. Trace each branching path: conditionals, switches, early returns, guard clauses, loops, error handlers. Trace each boundary condition: null, undefined, empty, zero, negative, overflow, max-length, type coercion, concurrency, timing. Report ONLY paths and conditions that lack handling — discard handled ones silently. Do NOT editorialize or add filler — findings only.**
+
+
+## EXECUTION
+
+### Step 1: Receive Content
+
+- Load the content to review strictly from provided input
+- If content is empty, or cannot be decoded as text, return empty array `[]` and stop
+- Identify content type (diff, full file, or function) to determine scope rules
```
[the remark this answers is not in the document]
Comment (alexeyv) [reply src/core/tasks/bmad-review-edge-case-hunter/workflow.md:23]: Fixed in 8be8123f — empty/bad input now returns a sentinel finding object instead of an empty array, so failures aren't silently suppressed.

File: src/core/tasks/bmad-review-edge-case-hunter/workflow.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+### Step 2: Exhaustive Path Analysis
+
+**Walk every branching path and boundary condition within scope — report only unhandled ones.**
+
+- If `also_consider` input was provided, incorporate those areas into the analysis
+- Enumerate all branching paths and boundary conditions within scope: conditionals, switches, early returns, guard clauses, loops, error handlers, null/empty states, overflow, type edges, concurrency, timing
```
[the remark this answers is not in the document]
Comment (alexeyv) [reply src/core/tasks/bmad-review-edge-case-hunter/workflow.md]: Rejected — also_consider is a focus-area hint, not a scope modifier. No ambiguity with diff mode scope rules.

File: src/core/tasks/bmad-review-edge-case-hunter/workflow.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+- For each path: determine whether the content handles it
+- Collect only the unhandled paths as findings — discard handled ones silently
+
+### Step 3: Validate Completeness
+
+- Recheck every conditional for missing else/default
+- Recheck every input for null/empty/wrong-type
+- Recheck loop bounds for off-by-one and empty-collection
+- Add any newly found unhandled paths to findings; discard confirmed-handled ones
```
[the remark this answers is not in the document]
Comment (alexeyv) [reply src/core/tasks/bmad-review-edge-case-hunter/workflow.md]: Fixed in 8be8123f — reframed Step 2 with two-lens approach (control flow + domain boundaries) and "derive from content" guidance. Step 3 now references Step 2 instead of duplicating a list.

File: src/core/tasks/bmad-skill-manifest.yaml [hunk 9]
```diff
@@ -18,9 +18,9 @@ index-docs.xml:
   type: task
   description: "Generates or updates an index.md to reference all docs in the folder"
 
-review-edge-case-hunter.xml:
+bmad-review-edge-case-hunter:
   canonicalId: bmad-review-edge-case-hunter
-  type: task
+  type: skill
   description: "Walk every branching path and boundary condition in content, report only unhandled edge cases"
```
[the remark this answers is not in the document]
Comment (alexeyv) [reply src/core/tasks/bmad-skill-manifest.yaml]: Dismissed — removed the entire entry from the central manifest. The skill has its own bmad-skill-manifest.yaml in its directory, so no alias is needed.
